chore: fix revealed addresses loose by token master

mprakhov opened the pull request:
### What does the PR do

1. If TokenMaster restores his account by a seed phrase - send him requests to join with revealed addresses
2. If a member edits the shared address, the token master receives it only when a control node approves the new addresses
3. Filter our outdated edit shared address and privileged user sync messages
4. Fixed changing privileged permissions (TM -> admin, admin -> TM)

status-go PR: https://github.com/status-im/status-go/pull/5383

Closes: https://github.com/status-im/status-desktop/issues/15139

### Affected areas

Sharing revealed addresses

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 9e9ccfb3 | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-15356/1/) | 2024-06-27 14:53:04 | ~6 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-15356/1//consoleText) |

anastasiyaig commented:
@mprakhov yes, sure, lets give it a spin tomorrow @virginiabalducci

virginiabalducci commented:
Hi @mprakhov
I'm seeing some issues in this build

1. Owner token and TMaster tokens are stuck at "Minting" status. I downloaded the build from this PR, created a community but I cannot mint an owner nor a tokenmaster tokens. 

![Screenshot 2024-07-01 at 12 21 21 PM](https://github.com/status-im/status-desktop/assets/68916150/5155c37f-e5ed-4894-858f-26e9e2743657)
As per https://sepolia.etherscan.io//tx/0x65cecc720a5b5e0bf22681334b31ebaef124706538c757e24b5072d6e5619ce0 it seems this is a cero balance transaction.

I created a different community and tried minting an owner/TM tokens using Optimism layer but the flow gets stuck loading the fees.
![Screenshot 2024-07-01 at 5 28 30 PM](https://github.com/status-im/status-desktop/assets/68916150/89172f0a-471b-498d-bd3d-f91815b64af4)

2. Also, I've airdropped (via Optimism) a tokenmaster from a different community to the same test user but user from the build of this PR is unable to access the tokenmaster permissions.

As we discussed via Discord, I'll be waiting for a rebase from the latest main build so I can continue testing. Thanks!

jrainville commented:
@mprakhov did you close by accident?

jrainville commented:
I can't reopen because it says there are no new commits :thinking:

Commit:
fix(wallet)_: SendModel scrollbar overlaps with tokens

- Adjusted StatusListItem height to account for scrollbar visibility by adding 16 pixels to contentHeight when tags are visible.
- Set tagsScrollBarVisible to true in TokenListView to ensure the scrollbar is displayed as needed.

fixes: 13620

virginiabalducci commented:
Scenarios tested:

1. TokenMaster joins the community, where there are some members already and he sees this member in the airdrop list -> QA passed

2. TM sees a new member in the airdrop list when the control node accepts him.
I tested this scenario with build https://github.com/status-im/status-desktop/commit/b61318474e2f0f1f97e20ae7aa1b92dc9f48cdae and as a Token Master I couldn't see the newly acepted members in the airdrop list. I created a new user for the latest build https://github.com/status-im/status-desktop/commit/63809d1c5b864a8b0fd199cb698014bb195fdd8b, user `35test` and I was able to see this member. However, the 2 previous users created on the previous build still do not show on the Airdrop list of the Token Master. All users show on the Owner (control node) list. 

https://github.com/status-im/status-desktop/assets/68916150/79a12806-ae64-4a09-bd17-963dffae2d5a

3. Member edit shared address, TM will airdrop to the new address (control node must be online)
QA tested and passed

**4. TM restore account from backup and should see members in the airdrop list (control node must be online to get members shared addresses)**
Member list is incomplete. The member added on the scenario 2, user `35test` which was previously seen in the tokenmaster's airdrop member list is not showing. You can compare the member list of the scenario 2's screen recording with these screenshots:
Tokenmaster view:
![Screenshot 2024-07-05 at 5 04 29 PM](https://github.com/status-im/status-desktop/assets/68916150/7f663656-1ccc-4b3f-853b-a83ce4b198ef)
![Screenshot 2024-07-05 at 5 04 38 PM](https://github.com/status-im/status-desktop/assets/68916150/bcb80409-991f-4855-b799-ad69a3ede2e3)

Owner (control node) view
![Screenshot 2024-07-05 at 5 05 25 PM](https://github.com/status-im/status-desktop/assets/68916150/3f20982a-1848-443d-99ac-1fa9b19e9173)
![Screenshot 2024-07-05 at 5 05 33 PM](https://github.com/status-im/status-desktop/assets/68916150/491b7a6b-9158-4393-aad8-e8a0f800c93c)

@mprakhov  Could you please check issue on scenario 4 ? 
Also, for the missing members from the airdrop list from scenario 2 that failed in the previous build, but correctly show up on new build, is there something that can be done for the missing users to display on the TM's airdrop list? 

Thanks!

virginiabalducci commented:
Hi @mprakhov, last Friday when I restored the TM profile for the first time, some members missing from the Airdrop list, however today when I opened the same account I was able to see the complete member list. To be on the safe side, I restored the account a second time, and I could confirm that the member list is displaying complete. 
With this last test, I can verify that Scenario 4 passed QA testing.

virginiabalducci commented:
@mprakhov yes correct all scenarios were tested and passed

jrainville commented:
@mprakhov can this PR be closed or the QML changes are needed?
